### ISSUE TITLE
 "is missing in props validation Error" occured when importing types

### ISSUE THREAD

**Issue by dkmqflx** [jsx-eslint/eslint-plugin-react#3278]
```ts
// types.ts

export type modalType = {
  title: string;
  visible: boolean;
  onCancel: () => void;
};

```

```ts
// useModal.ts
// hooks for open, close modal

import { useState } from 'react';

const useModal = () => {
  const [visible, setVisible] = useState<boolean>(false);

  const openModal = () => {
    setVisible(true);
  };

  const closeModal = () => {
    setVisible(false);
  };

  return { visible, setVisible, openModal, closeModal };
};

export default useModal;



```

```tsx
// Component that open Modal Component
// and pass props

import useModal from '../../hooks/useModal';
import {  Button } from 'antd';

const ProjectMasterTable = () => {
  const { visible, openModal, closeModal } = useModal();

  return (
    <>
          <Button type="primary" onClick={() => openModal()}>
            button
          </Button>

      {visible && (
        <ProjectMasterAddModal title="new" visible={visible} onCancel={closeModal}></ProjectMasterAddModal>
      )}
    </>
  );
};

export default ProjectMasterTable;


```

```tsx
// Components get props and throw erros

import { Modal, Form, Input, Radio } from 'antd';
import { modalType } from '../common/types';

const ProjectMasterAddModal = ({ title, visible, onCancel }: modalType) => {

  return (
    <Modal title={title} visible={visible} onCancel={onCancel}>   
    </Modal>
  );
};

export default ProjectMasterAddModal;
```
From above codes, I got two Errors `'visible' is missing in props validation`, `'onCancel' is missing in props validation`.
By using  `// eslint-disable-next-line react/prop-types` options, the error was disappeared, it works fine.
But this error don't occur whan i declare types in compnent like below

```tsx
// Components get props and do not throw error

import { Modal, Form, Input, Radio } from 'antd';

type modalType = {
  title: string;
  visible: boolean;
  onCancel: () => void;
};

const ProjectMasterAddModal = ({ title, visible, onCancel }: modalType) => {

  return (
    <Modal title={title} visible={visible} onCancel={onCancel}>   
    </Modal>
  );
};

export default ProjectMasterAddModal;
```

Additionally, when i used same hooks and types that import from other files, another component like below, any error do not occured.
So, i can't understand why i got this error
Is this because of importing type from other files or any other reasons ?

```tsx
// Component that open Modal Component
// and pass props

import { LoginPagerWrapper, LoginSection } from './styled';
import { modalType } from '../common/types';

const LoginPage = () => {
  const { visible, openModal, closeModal } = useModal();

  return (
    <>
      <LoginPagerWrapper>
        <LoginSection>
        </LoginSection>
      </LoginPagerWrapper>

      {visible && <SignUpModal title="join" visible={visible} onCancel={closeModal}></SignUpModal>}
    </>
  );
};

export default LoginPage;

```

```tsx
// Components get props and do not throw error

import { Modal } from 'antd';
import { modalType } from '../common/types';

const SignUpModal = ({ title, visible, onCancel }: modalType) => {
  const [form] = Form.useForm();

  return (
    <Modal title={title} visible={visible}  onCancel={onCancel}>

    </Modal>
  );
};

export default SignUpModal;


```

**Comment by dkmqflx** [comment]
> 

This is my folder structure, if so, why i do not get any errors in `LoginPage ` and `SignUpModal ` component.
It import types from common/types.ts that types is not declared in same file.

- components
  - common
     - types.ts
  - login
  -   SignUpModal.tsx
  - project
    - ProjectMasterAddModal.tsx
    - ProjectMasterTable.tsx
    
- pages
  - LoginPage.tsx

**Comment by ljharb** [comment]
I'm not really sure without looking at the actual code.

If you can provide a repro repo, I'm happy to reopen this.
